### PR TITLE
add: android sdk init

### PR DESCRIPTION
**Pull request by mauriguz**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

About # (link your issue here)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: android/src/main/java/org/gooddollar/FaceTecModule.java [hunk 1]
```diff
@@ -135,29 +135,35 @@ public String getName() {
...
+            case DEVICE_IN_REVERSE_PORTRAIT_MODE:
+                FaceTecSDK.setDynamicStrings(Customization.UITextStrings);
+                promise.resolve(true);
+
+                break;
+            case NEVER_INITIALIZED:
+            case NETWORK_ISSUES:
+                FaceVerification.register(serverURL, jwtAccessToken);
+
+                if (!licenseText.isEmpty()) {
```
Comment (serdiukov-o-nordwhale) [inline android/src/main/java/org/gooddollar/FaceTecModule.java:156]: licenseText could also be null

File: android/src/main/java/org/gooddollar/FaceTecModule.java [hunk 2]
```diff
@@ -177,4 +183,29 @@ public void faceVerification(String enrollmentIdentifier,
...
+            public void onCompletion(final boolean successful) {
+                if(successful) {
+                    FaceTecSDK.setDynamicStrings(Customization.UITextStrings);
+                    promise.resolve(true);
+                    return;
+                }
+
+                FaceTecSDKStatus sdkStatus = getSDKStatus(activity);
+                String customMessage = null;
+                if (sdkStatus == FaceTecSDKStatus.NEVER_INITIALIZED) {
```
Comment (serdiukov-o-nordwhale) [inline android/src/main/java/org/gooddollar/FaceTecModule.java:203]: 1. make it multi-line like on web & IOS
2. replace word "Simulator" to "Android Emulator"

Comment (serdiukov-o-nordwhale) [inline android/src/main/java/org/gooddollar/FaceTecModule.java:203]:
```suggestion

                if (sdkStatus == FaceTecSDKStatus.NEVER_INITIALIZED) {
```

File: android/src/main/java/org/gooddollar/FaceTecModule.java [hunk 3]
```diff
@@ -135,29 +135,35 @@ public String getName() {
...
+        final Activity activity = getCurrentActivity();
+        FaceTecSDKStatus status = getSDKStatus(activity);
+
+        switch (status) {
+            case INITIALIZED:
+            case DEVICE_IN_LANDSCAPE_MODE:
+            case DEVICE_IN_REVERSE_PORTRAIT_MODE:
+                FaceTecSDK.setDynamicStrings(Customization.UITextStrings);
+                promise.resolve(true);
+
```
Comment (serdiukov-o-nordwhale) [inline android/src/main/java/org/gooddollar/FaceTecModule.java:150]:
```suggestion
```

File: android/src/main/java/org/gooddollar/FaceTecModule.java [hunk 4]
```diff
@@ -135,29 +135,35 @@ public String getName() {
...
+            case NETWORK_ISSUES:
+                FaceVerification.register(serverURL, jwtAccessToken);
+
+                if (!licenseText.isEmpty()) {
+                    FaceTecSDK.initializeInProductionMode(activity, licenseText, licenseKey, encryptionKey, onInitializationAttempt(activity, promise));
+                    return;
+                }
+
+                FaceTecSDK.initializeInDevelopmentMode(activity, licenseKey, encryptionKey, onInitializationAttempt(activity, promise));
+
```
Comment (serdiukov-o-nordwhale) [inline android/src/main/java/org/gooddollar/FaceTecModule.java:162]:
```suggestion
```

File: android/src/main/java/org/gooddollar/FaceTecModule.java [hunk 5]
```diff
@@ -177,4 +183,29 @@ public void faceVerification(String enrollmentIdentifier,
         lastProcessor = processor;
         processor.enroll(enrollmentIdentifier, maxRetries);
     }
+
+    private FaceTecSDKStatus getSDKStatus(Activity activity) {
```
Comment (serdiukov-o-nordwhale) [inline android/src/main/java/org/gooddollar/FaceTecModule.java:187]: it seems we don't need  a separate method
on IOS implementation getStatus is needed to be wrapped onto the main thread
if you don't need additional logic here, just use FaceTecSDK.getStatus(activity) everywhere

**Review by serdiukov-o-nordwhale — CHANGES_REQUESTED** [review]

**Comment by serdiukov-o-nordwhale** [comment]
Please also append your code with the comments describing the flow

File: android/src/main/java/org/gooddollar/FaceTecModule.java [hunk 6]
```diff
@@ -177,4 +183,31 @@ public void faceVerification(String enrollmentIdentifier,
...
+                    return;
+                }
+
+                FaceTecSDKStatus sdkStatus = FaceTecSDK.getStatus(activity);
+                String newLine = System.getProperty("line.separator");
+                String customMessage = null;
+
+                // if status still hasn't been initialized it means user is using an emulator
+                if (sdkStatus == FaceTecSDKStatus.NEVER_INITIALIZED) {
+                    customMessage = "Initialize wasn't attempted as Android Emulator has been detected." + newLine + "FaceTec SDK could be ran on the real devices only";
```
Comment (serdiukov-o-nordwhale) [inline android/src/main/java/org/gooddollar/FaceTecModule.java:206]:
```suggestion
                    customMessage = "Initialize wasn't attempted as Android Emulator has been detected."
                        + "FaceTec SDK could be ran on the real devices only";
```

**Review by serdiukov-o-nordwhale — APPROVED** [review]

File: android/src/main/java/org/gooddollar/FaceTecModule.java [hunk 7]
```diff
@@ -177,4 +183,32 @@ public void faceVerification(String enrollmentIdentifier,
...
+                // the value of successful determines if the sdk has been initialized or not
+                if(successful) {
+                    // status is already initialized - resolve promise with true
+                    FaceTecSDK.setDynamicStrings(Customization.UITextStrings);
+                    promise.resolve(true);
+                    return;
+                }
+
+                FaceTecSDKStatus sdkStatus = FaceTecSDK.getStatus(activity);
+                String newLine = System.getProperty("line.separator");
```
Comment (mauriguz) [inline android/src/main/java/org/gooddollar/FaceTecModule.java:201]:
```suggestion

```

Comment (serdiukov-o-nordwhale) [inline android/src/main/java/org/gooddollar/FaceTecModule.java]:
```suggestion
```

**Review by serdiukov-o-nordwhale — APPROVED** [review]

**Pull request merged**